pkg/services/ngalert/schedule/schedule_test.go

```diff
@@ -222,6 +227,79 @@ func TestAlertingTicker(t *testing.T) {
 	})
 }
 
+func TestSendingToExternalAlertmanager(t *testing.T) {
```

Comment:
I feel pretty confident with this test, hence I didn't go for an integration test which would be testing more of that same. Once I have the APIs for discovered/dropped alertmanager(s) I'll make sure to create one that covers this.

Reply:
We are missing few cases here that could fall between existing 2 cases that we test already
1. More than 1 org (ideally adding 2nd org after 1st org was already present)
2. Removing few orgs when there are more than 1 org (and not removing all the orgs, this is already covered)
2. Modifying the admin config while there exists one already for the org, so make sure the hash check is working fine.

If there is any way to know if a org's sender was updated or not, then that could help test the case where we don't re-apply config unnecessarily (look at my other comment on this part). But it is not a blocker from my side.

Reply:
I should have addressed all of these cases now - WDYT?